### PR TITLE
shim-unsigned-aarch64: upgrade to 15.4 with revert Revert "shim-unsigned: Force using shim-15 for aarch64 (#880)"

### PR DESCRIPTION
**Pull request by anphel31**
This reverts commit cbfa4906440394ea5dcd53beac857ed5605453fc.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade `shim-unsigned-aarch64` to version 15.4, which resolves build breaks with 3.0 toolchain. This reverts PR: #880 
"shim-unsigned: Force using shim-15 for aarch64 (#880)"

Version 15 has a number of errors similar to the following:
```
"gcc -ggdb -O0 -I/usr/src/mariner/BUILD/shim-15/Cryptlib -iquote /usr/src/mariner/BUILD/shim-15/Cryptlib -fno-stack-protector -fno-strict-aliasing -fpic -fshort-wchar -Wall -I/usr/src/mariner/BUILD/shim-15/Cryptlib/Include -I/usr/include/efi -I/usr/include/efi/aarch64 -I/usr/include/efi/protocol -std=gnu89 -ffreestanding -I/usr/lib/gcc/aarch64-unknown-linux-gnu/13.2.0/include -DMDE_CPU_AARCH64   -c -o Hash/CryptMd4Null.o Hash/CryptMd4Null.c"
"In file included from /usr/include/efi/efilib.h:22,"
"                 from /usr/src/mariner/BUILD/shim-15/Cryptlib/Include/OpenSslSupport.h:19,"
"                 from /usr/src/mariner/BUILD/shim-15/Cryptlib/Include/stddef.h:15,"
"                 from /usr/include/efi/aarch64/efibind.h:41,"
"                 from /usr/include/efi/efi.h:48,"
"                 from /usr/src/mariner/BUILD/shim-15/Cryptlib/Library/BaseLib.h:1,"
"                 from /usr/src/mariner/BUILD/shim-15/Cryptlib/InternalCryptLib.h:18,"
"                 from Hash/CryptMd4Null.c:15:"
"/usr/include/efi/efidebug.h:22:8: error: unknown type name 'UINTN'"
"   22 | extern UINTN     EFIDebug;"
"      |        ^~~~~"
"/usr/include/efi/efidebug.h:95:1: error: unknown type name 'INTN'"
"   95 | INTN"
"      | ^~~~"

```

###### Change Log  

- Revert PR 880

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build https://dev.azure.com/mariner-org/mariner/_build/results?buildId=455647&view=results

### CODE REVIEW

**Review by christopherco — COMMENTED** [review]
Revert itself looks good - glad to see that old commit get used 😄 
Can you add what the 3.0 toolchain build break error was to the PR description? I'm also curious if it was related to the recent `gnu-efi` upgrade

**Comment by anphel31** [comment]
> Revert itself looks good - glad to see that old commit get reverted finally 😄 Can you add what the 3.0 toolchain build break error was to the PR description? I'm also curious if it was related to the recent `gnu-efi` upgrade

Sure, I've added one of the errors to the description (there are many of the same type of errors in the full log). I believe it's caused by missing the include for `stdint.h` in the right place, and possibly fixed in 15.3 with [this change](https://github.com/rhboot/shim/commit/766aac4d5cfbe76026be5ce718b0883ee211f323)

**Review by tobiasb-ms — APPROVED** [review]

File: SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec [hunk 1]
```diff
@@ -47,6 +47,7 @@ make VENDOR_CERT_FILE=cert.der test
 * Tue Mar 30 2021 Chris Co <chrco@microsoft.com> - 15.4-1
 - Update to 15.4
 - Remove extra patches. These are incorporated into latest version
+- License verified
```
Comment (christopherco) [inline SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec:50]: I approve of this addition into my original changelog entry

File: SPECS/shim-unsigned/shim-unsigned.spec [hunk 2]
```diff
@@ -41,6 +41,7 @@ install -D -m 0744 %{_datadir}/shim-unsigned-aarch64/shimaa64.efi %{buildroot}/b
 %changelog
 * Tue Mar 30 2021 Chris Co <chrco@microsoft.com> - 15.4-1
 - Update to 15.4
+- License verified
```
Comment (christopherco) [inline SPECS/shim-unsigned/shim-unsigned.spec:44]: I approve of this addition into my original changelog entry

**Comment by christopherco** [comment]
Adding my S-O-B for the changelog entries

Signed-off-by: Chris Co <chrco@microsoft.com>

**Pull request merged**